image_proc/src/debayer.cpp

```diff
@@ -31,95 +31,73 @@
 #include <opencv2/imgproc/imgproc.hpp>
 // Until merged into OpenCV
-#include "edge_aware.h"
 
+#include <rcutils/cmdline_parser.h>
 #include <cv_bridge/cv_bridge.h>
-
-namespace image_proc {
-
-namespace enc = sensor_msgs::image_encodings;
-
-class DebayerNodelet : public nodelet::Nodelet
+#include "debayer.hpp"
+namespace image_proc
 {
-  // ROS communication
-  boost::shared_ptr<image_transport::ImageTransport> it_;
-  image_transport::Subscriber sub_raw_;
-  
-  boost::mutex connect_mutex_;
-  image_transport::Publisher pub_mono_;
-  image_transport::Publisher pub_color_;
-
-  // Dynamic reconfigure
-  boost::recursive_mutex config_mutex_;
-  typedef image_proc::DebayerConfig Config;
-  typedef dynamic_reconfigure::Server<Config> ReconfigureServer;
-  boost::shared_ptr<ReconfigureServer> reconfigure_server_;
-  Config config_;
-
-  virtual void onInit();
-
-  void connectCb();
-
-  void imageCb(const sensor_msgs::ImageConstPtr& raw_msg);
-
-  void configCb(Config &config, uint32_t level);
-};
-
-void DebayerNodelet::onInit()
+namespace enc = sensor_msgs::image_encodings;
+DebayerNode::DebayerNode(const rclcpp::NodeOptions& options)
+: Node("DebayerNode", options)
 {
-  ros::NodeHandle &nh         = getNodeHandle();
-  ros::NodeHandle &private_nh = getPrivateNodeHandle();
-  it_.reset(new image_transport::ImageTransport(nh));
+  auto parameter_change_cb =
+    [this](std::vector<rclcpp::Parameter> parameters) -> rcl_interfaces::msg::SetParametersResult
+    {
+      auto result = rcl_interfaces::msg::SetParametersResult();
+      result.successful = true;
+      for (auto parameter : parameters) {
+        if (parameter.get_name() == "camera_namespace") {
+          camera_namespace_ = parameter.as_string();
+          RCLCPP_INFO(get_logger(), "reset camera_namespace to %s! ", camera_namespace_.c_str());
+        }
+      std::string image_mono = camera_namespace_+"/image_mono";
+      std::string image_color = camera_namespace_+"/image_color";
+      connectCb();
+      std::lock_guard<std::mutex> lock(connect_mutex_);
+      RCLCPP_INFO(this->get_logger(), "mono: %s, color: %s", image_mono.c_str(), image_color.c_str());
+      pub_mono_  = image_transport::create_publisher(this, image_mono);
+      pub_color_ = image_transport::create_publisher(this, image_color);
+      }
+      return result;
+  };
 
-  // Set up dynamic reconfigure
-  reconfigure_server_.reset(new ReconfigureServer(config_mutex_, private_nh));
-  ReconfigureServer::CallbackType f = boost::bind(&DebayerNodelet::configCb, this, _1, _2);
-  reconfigure_server_->setCallback(f);
 
-  // Monitor whether anyone is subscribed to the output
-  typedef image_transport::SubscriberStatusCallback ConnectCB;
-  ConnectCB connect_cb = boost::bind(&DebayerNodelet::connectCb, this);
+  this->set_on_parameters_set_callback(parameter_change_cb);
   // Make sure we don't enter connectCb() between advertising and assigning to pub_XXX
-  boost::lock_guard<boost::mutex> lock(connect_mutex_);
```

Comment:
This should probably not be removed ? Otherwise it will not start publishing unless a reconfiguration request is made. Let me know if I'm missing something here, but will add until I hear otherwise.